src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+  const {
+    style: tooltipStyle,
+    className: tooltipClassName,
+    ...tooltipOpts
+  } = tooltipOptsProp;
+
+  const tooltipClassNames = classNames(
+    styles.Tooltip,
+    styles[direction],
+    tooltipClassName,
+    {
+      [styles.open]: isOpen || hovered,
+    }
+  );
+
+  const offsets = useMemo(() => {
+    // Properly center tooltips on the left and right to the center
+    // of the actual element we are attaching the tooltip.
+    if (direction === 'left' || direction === 'right') {
+      return {
+        marginTop: -(tooltipHeight / 2),
+      };
+    }
+
+    return {};
+  });
+
+  const isText = typeof tooltipContent === 'string';
+
+  const tooltipWrapperProps = {
+    style: {
+      width: (tooltipStyle && tooltipStyle.width) || tooltipWidth,
+      ...offsets,
+    },
+    className: tooltipClassNames,
+  };
+
+  const TooltipContent = tooltipContent && (
+    <div
+      id={tooltipId}
+      aria-hidden={isOpen}
+      ref={tooltipRef}
+      {...tooltipWrapperProps}
+    >
+      <div
+        style={tooltipStyle}
+        className={styles.tooltipInnerWrapper}
+        {...tooltipOpts}
+      >
+        {!!closeable && (
+          <button className={styles.tooltipClose} onClick={closeFunc}>
+            <Icon name="close" />
+          </button>
+        )}
+
+        {isText ? (
+          <div className={styles.textContent}>{tooltipContent}</div>
+        ) : (
+          tooltipContent
+        )}
+      </div>
+    </div>
+  );
+
+  return (
+    <div className={styles.wrapper} {...rest}>
+      {TooltipContent}
+      <div
+        aria-describedby={tooltipId}
+        onMouseEnter={handleSetHover(true)}
+        onMouseLeave={handleSetHover(false)}
+        className={styles.original}
+      >
+        {children}
+      </div>
+    </div>
+  );
+};
+
+Tooltip.propTypes = {
+  children: PropTypes.node.isRequired,
+  closeable: PropTypes.func,
+  closeOnEscape: PropTypes.bool,
+  content: PropTypes.node,
+  direction: PropTypes.oneOf(directions),
+  showOnHover: PropTypes.bool,
+  isOpen: PropTypes.bool,
+  tooltipProps: PropTypes.shape({
+    className: PropTypes.string,
+    style: PropTypes.object,
+  }),
+  tooltipWidth: PropTypes.string,
+};
+
+Tooltip.defaultProps = {
+  content: null,
+  closeable: null,
+  closeOnEscape: true,
+  direction: 'top',
+  showOnHover: false,
+  isOpen: false,
+  tooltipProps: {},
+  tooltipWidth: null,
+};
+
+export default Tooltip;
+
+export const EasyRichTooltip = ({ closeable, children, ...props }) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const open = () => setOpen(true);
+  const close = () => setOpen(false);
+  const toggle = () => setOpen(prev => !prev);
+
+  return (
+    <Tooltip
+      isOpen={isOpen}
+      closeable={() => {
+        if (closeable) {
+          closeable();
+        }
+        setOpen(false);
+      }}
+      {...props}
+    >
+      {children({ open, close, toggle })}
+    </Tooltip>
+  );
+};
+
+EasyRichTooltip.propTypes = {
```

Comment:
Are you planning for a non-rich version? If not, would `EasyTooltip` work?

Reply:
EasyTooltip would be just for text, and just with text can be done just using `Tooltip` by itself. All of the other props are _mostly_ to make the rich content and interaction possible